akka-cluster-metrics/src/main/scala/akka/cluster/metrics/ClusterMetricsCollector.scala

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.cluster.metrics
+
+import akka.actor.Actor
+import akka.actor.ActorLogging
+import akka.actor.Props
+import akka.actor.Address
+import akka.cluster.InternalClusterAction
+import akka.cluster.ClusterEvent
+import akka.cluster.Member
+import akka.cluster.Cluster
+import scala.collection.immutable
+import akka.cluster.MemberStatus
+import scala.concurrent.forkjoin.ThreadLocalRandom
+import akka.actor.Terminated
+
+/**
+ *  Runtime collection management commands.
+ */
+sealed abstract class CollectionControlMessage extends Serializable
+
+/**
+ * Command for [[ClusterMetricsSupervisor]] to start metrics collection.
+ */
+@SerialVersionUID(1L)
+case object CollectionStartMessage extends CollectionControlMessage {
+  /** Java API */
+  def getInstance = CollectionStartMessage
+}
+
+/**
+ * Command for [[ClusterMetricsSupervisor]] to stop metrics collection.
+ */
+@SerialVersionUID(1L)
+case object CollectionStopMessage extends CollectionControlMessage {
+  /** Java API */
+  def getInstance = CollectionStopMessage
+}
+
+/**
+ * INTERNAL API.
+ *
+ * Actor providing customizable metrics collection supervision.
+ */
+private[metrics] class ClusterMetricsSupervisor extends Actor with ActorLogging {
+  import ClusterMetricsExtension._
+  val metrics = ClusterMetricsExtension(context.system)
+  import metrics.settings._
+  import context._
+
+  override val supervisorStrategy = metrics.strategy
+
+  var collectorInstance = 0
+
+  def collectorName = s"collector-${collectorInstance}"
+
+  override def preStart() = {
+    if (CollectorEnabled) {
+      self ! CollectionStartMessage
+    } else {
+      log.warning(s"Metrics collection is disabled in configuration. Use subtypes of ${classOf[CollectionControlMessage].getName} to manage collection at runtime.")
+    }
+  }
+
+  override def receive = {
+    case CollectionStartMessage ⇒
+      children.foreach(stop)
+      collectorInstance += 1
+      actorOf(Props(classOf[ClusterMetricsCollector]), collectorName)
+      log.debug(s"Collection started.")
+    case CollectionStopMessage ⇒
+      children.foreach(stop)
+      log.debug(s"Collection stopped.")
+  }
+
+}
+
+/**
+ * Local cluster metrics extension events.
+ *
+ * Published to local event bus subscribers by [[ClusterMetricsCollector]].
+ */
+trait ClusterMetricsEvent
+
+/**
+ * Current snapshot of cluster node metrics.
+ */
+final case class ClusterMetricsUpdated(nodeMetrics: Set[NodeMetrics]) extends ClusterMetricsEvent {
```

Comment:
if we don't have a good reason it makes it somewhat easier for users to migrate if the same name is used, so yes please